beacon-chain/core/helpers/slot_epoch.go

```diff
@@ -82,6 +82,16 @@ func StartSlot(epoch uint64) (uint64, error) {
 	return slot, nil
 }
 
+// EndSlot returns the last slot number of the
+// current epoch.
+func EndSlot(epoch uint64) (uint64, error) {
+	slot, err := mathutil.Mul64(epoch+1, params.BeaconConfig().SlotsPerEpoch)
```

Comment:
It is extremely unlikely, but if someone passed maxUint64 as the argument, then this would end up with slot == 0.
```suggestion
	if epoch == math.MaxUint64 {
		return 0, errors.New("start slot calculation overflows")
	}
	slot, err := mathutil.Mul64(epoch+1, params.BeaconConfig().SlotsPerEpoch)
```

(Please add regression test to confirm)